### PR TITLE
[PyOV] Add Python API for MaxPool-14 and AvgPool-14

### PR DESCRIPTION
**Pull request by p-wysocki**
### Details:
 - Extend Python API with`MaxPool-14` and `AvgPool-14`
 - They both introduce a new ceil mode: `ov::op::RoundingType::CEIL_TORCH`
 - The new ceiling mode does not allow the last pooling in a Dimension to start in the padding area

### Related PRs
 - https://github.com/openvinotoolkit/openvino/pull/22930
 - https://github.com/openvinotoolkit/openvino/pull/22796
 - https://github.com/openvinotoolkit/openvino/pull/23027
 - https://github.com/openvinotoolkit/openvino/pull/23381
 - https://github.com/openvinotoolkit/openvino/pull/23582

### Tickets:
 - 131961

### Context
https://github.com/openvinotoolkit/openvino/issues/18731

### CODE REVIEW

File: src/bindings/python/tests/test_graph/test_pooling.py [hunk 1]
```diff
@@ -32,6 +32,66 @@ def test_avg_pool_2d(ndarray_1x1x4x4):
...
+    pads_end = [1, 1]
+
+    node = ov.avg_pool(param, strides, pads_begin, pads_end, kernel_shape, False, "ceil_torch")
+    assert node.get_type_name() == "AvgPool"
+    assert node.get_output_size() == 1
+    assert list(node.get_output_shape(0)) == [1, 3, 5, 5]
+    assert node.get_output_element_type(0) == Type.f32
+
+
+def test_avg_pool_3x3_kernel_ceil_torch_strides():
```
Comment (akuporos) [inline src/bindings/python/tests/test_graph/test_pooling.py:65]: some of these tests looks pretty similar with different values.
May be use parametrization to reduce the number of repeated checks?

Comment (p-wysocki) [reply src/bindings/python/tests/test_graph/test_pooling.py]: Done, the tests have been parametrized.

File: src/bindings/python/src/openvino/runtime/opset14/ops.py [hunk 2]
```diff
@@ -62,3 +65,103 @@ def inverse(
...
+def max_pool(
+    data: NodeInput,
+    strides: List[int],
+    dilations: List[int],
+    pads_begin: List[int],
+    pads_end: List[int],
+    kernel_shape: TensorShape,
+    rounding_type: str = "floor",
+    auto_pad: Optional[str] = None,
+    index_element_type: Optional[str] = "i64",
```
Comment (mitruska) [inline src/bindings/python/src/openvino/runtime/opset14/ops.py:80]: Can `ov.Type` be allowed as well? 

```suggestion
    index_element_type: Optional[Union[Type, str]] = "i64",
```

File: src/bindings/python/src/openvino/runtime/opset14/ops.py [hunk 3]
```diff
@@ -62,3 +65,103 @@ def inverse(
...
+    rounding_type: str = "floor",
+    auto_pad: Optional[str] = None,
+    name: Optional[str] = None,
+) -> Node:
+    """Return average pooling node.
+
+    :param data_batch:      The input node providing data.
+    :param strides:         The window movement strides.
+    :param pads_begin:      The input data optional padding below filled with zeros.
+    :param pads_end:        The input data optional padding below filled with zeros.
```
Comment (mitruska) [inline src/bindings/python/src/openvino/runtime/opset14/ops.py:142]: Worth to align with the max_pool docs, as better described:
```suggestion
    :param pads_begin:      The number of pixels to add at the beginning along each axis.
    :param pads_end:        The number of pixels to add at the end along each axis.
```

**Review by mitruska — APPROVED** [review]

File: src/bindings/python/src/openvino/runtime/opset14/ops.py (hunk 2 above)
Comment (p-wysocki) [reply src/bindings/python/src/openvino/runtime/opset14/ops.py:80]: I don't think so, we would have to have an additional logic in the operator wrapper, which would either translate `ov.Type` from Python to C++. It's currently being passed "raw" into `NodeFactory`.

https://github.com/openvinotoolkit/openvino/blob/902f1b1af0f5abff87c32dca62d261ce5d1d01c2/src/bindings/python/src/openvino/runtime/opset8/ops.py#L304-L318

If you wish we can consider adding that logic. cc @akuporos @jiwaszki do you recall if we do this for other ops?

**Review by jiwaszki — APPROVED** [review]

File: src/bindings/python/src/openvino/runtime/opset14/ops.py (hunk 2 above)
Comment (p-wysocki) [reply src/bindings/python/src/openvino/runtime/opset14/ops.py]: It works with `ov.Type`. Change applied.

**Pull request merged**